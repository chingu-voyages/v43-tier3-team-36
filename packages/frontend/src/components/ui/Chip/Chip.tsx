/* eslint-disable react/jsx-props-no-spreading */
import type { ForwardRefExoticComponent, HTMLAttributes } from 'react';
import { forwardRef } from 'react';
import clsx from 'clsx';

export interface ChipProps extends HTMLAttributes<HTMLSpanElement> {
  /**
   * Label for the component
   */
  label: string;
  /**
   * Whether to make component size larger
   * @default false
   */
  large?: boolean;
}

const styles = {
  base: 'font-semibold rounded-full text-xs px-2 py-1 text-white bg-red-500',
};

export const Chip: ForwardRefExoticComponent<
ChipProps & React.RefAttributes<HTMLSpanElement>
> = forwardRef(({
  label, large = false, className, ...rest
}, ref) => {
  const spanClassNames = clsx(styles.base, { 'text-sm': large }, className);

  return (
    <span ref={ref} className={spanClassNames} {...rest}>
      {label}
    </span>
  );
});
