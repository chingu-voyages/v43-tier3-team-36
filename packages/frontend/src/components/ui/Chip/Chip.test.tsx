import { render } from 'test-utils';

import { Chip } from './Chip';

describe('<Chip/>', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<Chip label="Foo" />);
      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with large prop', () => {
      const { container } = render(<Chip label="Foo" large />);
      expect(container).toMatchSnapshot();
    });
  });
});
