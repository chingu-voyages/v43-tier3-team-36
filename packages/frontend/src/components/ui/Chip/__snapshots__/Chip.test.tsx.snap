// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chip/> Snapshots should match snapshot 1`] = `
<div>
  <span
    class="font-semibold rounded-full text-xs px-2 py-1 text-white bg-red-500"
  >
    Foo
  </span>
</div>
`;

exports[`<Chip/> Snapshots should match snapshot with large prop 1`] = `
<div>
  <span
    class="font-semibold rounded-full text-xs px-2 py-1 text-white bg-red-500 text-sm"
  >
    Foo
  </span>
</div>
`;
